Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        static Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            using var image = ApplicationServices.GetScreenshot();
+            var result = new ScreenshotResult(new NSImage(image, new CGSize(image.Width, image.Height)));
```

Comment:
Maybe we should do as little as possible here and pass CGImage to the result directly. Then do all the work in the get.

Reply:
Then we would have to dispose ScreenshotResult again, which is less desirable than disposing a stream